tests/integration/detection/test_model.py

```diff
@@ -45,6 +45,11 @@ def test__init() -> None:
     assert model == Model(name)
 
 
+def test__init__validate_name() -> None:
```

Comment:
Nit: this can be in unit-test, same comment for other similar tests.

Reply:
Will do for similar tests as well.